UI/MeterBrowseWnd.cpp

```diff
@@ -580,4 +585,360 @@ void ShipDamageBrowseWnd::UpdateEffectLabelsAndValues(GG::Y& top) {
+
+    GG::Label* bay_summary = new CUILabel(boost::io::str(FlexibleFormat(UserString("TT_BREAKDOWN_SUMMARY"))
+                                                         % UserString("SHIP_FIGHTER_BAY_SUMMARY") % bay_value_text));
+    bay_summary->SetFont(ClientUI::GetBoldFont());
+    bay_summary->MoveTo(GG::Pt(GG::X0, top));
+    bay_summary->Resize(GG::Pt(LABEL_WIDTH + VALUE_WIDTH, m_row_height));
+    AttachChild(bay_summary);
+
+    GG::Label* hangar_summary = new CUILabel(boost::io::str(FlexibleFormat(UserString("TT_BREAKDOWN_SUMMARY"))
+                                                            % UserString("SHIP_FIGHTER_HANGAR_SUMMARY") % hangar_value_text));
+    hangar_summary->SetFont(ClientUI::GetBoldFont());
+    hangar_summary->MoveTo(GG::Pt(ROW_WIDTH, top));
+    hangar_summary->Resize(GG::Pt(LABEL_WIDTH + VALUE_WIDTH, m_row_height));
+    AttachChild(hangar_summary);
+    m_effect_labels_and_values.push_back(std::make_pair(bay_summary, hangar_summary));
+
+    top += m_row_height;
+
+    std::string fighter_damage_text = ColourWrappedtext(DoubleToString(fighter_damage, 3, false), DAMAGE_COLOR);
+
+    if (!m_show_all_bouts) {
+        // Show damage for first wave (2nd combat round)
+        std::pair<int, float> first_wave = FightersDamageForBout(bay_total_capacity, hangar_current_fighters,
+                                                                    fighter_damage, 2);
+        std::string launch_text = ColourWrappedtext(boost::lexical_cast<std::string>(first_wave.first), HIGHLIGHT_COLOR);
+        std::string damage_label_text = boost::io::str(FlexibleFormat(UserString("TT_FIGHTER_DAMAGE"))
+                                                       % launch_text % fighter_damage_text);
+        GG::Label* damage_label = new CUILabel(damage_label_text, GG::FORMAT_RIGHT);
+        damage_label->MoveTo(GG::Pt(GG::X0, top));
+        damage_label->Resize(GG::Pt(LABEL_WIDTH * 2 + EDGE_PAD, m_row_height));
+        AttachChild(damage_label);
+
+        std::string damage_value_text = DoubleToString(first_wave.second, 3, false);
+        GG::Label* damage_value = new CUILabel(ColourWrappedtext(damage_value_text, DAMAGE_COLOR));
+        damage_value->MoveTo(GG::Pt(LABEL_WIDTH * 2, top));
+        damage_value->Resize(GG::Pt(VALUE_WIDTH * 2, m_row_height));
+        damage_value->SetFont(ClientUI::GetBoldFont());
+        AttachChild(damage_value);
+        m_effect_labels_and_values.push_back(std::make_pair(damage_label, damage_value));
+
+        top += m_row_height;
+    } else {
+        // add labels for bay parts
+        for (std::multimap<std::string, int>::value_type& bay_part : bay_parts) {
+            const std::string& part_name = UserString(bay_part.first);
+            int fighter_total = bay_part.second;
+            m_bay_list->Insert(new ShipFightersBrowseRow(part_name, fighter_total));
+        }
+
+        // add a label for each hangar part, distributing the total fighters equally
```

Comment:
could instead have "hangar_part x 6" and just the sum of fighters in that hangar part type...